juriscraper/state/texas/common.py

```diff
@@ -0,0 +1,462 @@
+                representatives=[
+                    clean_string(text)
+                    for text in parties["Representative"][i].xpath(".//text()")
+                ],
+            )
+            for i in range(n_parties)
+        ]
+
+    def _parse_trial_court(self) -> TexasTrialCourt:
+        """
+        Extracts the trial court info from the HTML tree. Will fail if `_parse_text`
+        has not yet been called.
+
+        :return: Trial court info.
+        """
+        info_panel: HtmlElement = self.tree.find(
+            './/*[@id="panelTrialCourtInfo"]/div[2]'
+        )
+        fields: dict[str, str] = {
+            clean_string(child.find(".//*[1]").text_content()): clean_string(
+                child.find(".//*[2]").text_content()
+            )
+            for child in info_panel.iterchildren()
+        }
+
+        return TexasTrialCourt(
+            name=fields.get("Court", ""),
+            county=fields.get("County", ""),
+            judge=fields.get("Court Judge", ""),
+            case=fields.get("Court Case", ""),
+            reporter=fields.get("Reporter", ""),
+            punishment=fields.get("Punishment", ""),
+        )
+
+    @staticmethod
+    def _parse_case_documents(cell: HtmlElement) -> list[TexasCaseDocument]:
+        """
+        Helper method to parse case documents for a given docket entry.
+
+        Tries to find a table in the given cell and extracts the document URLs and names from it. If no table is found, returns an empty list.
+
+        :return: List of case documents.
+        """
+        table = cell.find(".//table")
+        if table is None:
+            return []
+        documents = parse_table(table)
+        n = len(documents["0"])
+
+        return [
+            TexasCaseDocument(
+                url=documents["0"][i].find(".//a").get("href"),
+                name=clean_string(documents["1"][i].text_content()),
+            )
+            for i in range(n)
+        ]
+
+    def _parse_case_events(self) -> list[TexasCaseEvent]:
+        """
+        Extracts and parses case events.
+
+        This method processes the "Case Events" table and produces a list of
+        TexasCaseEvent objects. If the table is empty, an empty list is returned.
+
+        :return: A list of parsed TexasCaseEvent objects.
+        """
+        # Works because when there are no entries, Texas places a single <td> element, which will be parsed by parse_table as 1 entry in the first column and 0 in all others.
+        if len(self.events["Event Type"]) == 0:
+            return []
+        n = len(self.events["Date"])
+
+        return [
+            TexasCaseEvent(
+                date=datetime.strptime(
+                    clean_string(self.events["Date"][i].text_content()),
+                    self.date_format,
+                ),
+                type=clean_string(self.events["Event Type"][i].text_content()),
+                documents=self._parse_case_documents(
```

Comment:
Here and in `_parse_appellate_briefs`, can you please rename documents to attachments so it matches the naming used in our other scrapers?

```suggestion
                attachments=self._parse_case_documents(
```